internal/provider/data_source_gitlab_repository_file.go

```diff
@@ -0,0 +1,107 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerDataSource("gitlab_repository_file", func() *schema.Resource {
+	return &schema.Resource{
+		Description: "Allows you to receive information about file in repository like name, size, content. File content is Base64 encoded. This endpoint can be accessed without authentication if the repository is publicly accessible.",
```

Comment:
Please use the same description pattern as here:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/c6a706e3ab6e907e2da0c92ee50138544ff501dd/internal/provider/data_source_gitlab_project_issue.go#L13

w.r.t to wording and the upstream docs link

Reply:
I will take a look again probably on Sunday. I had to actually implement something at my workplace :D